lib/rucio/transfertool/fts3.py

```diff
@@ -410,44 +487,88 @@ def bulk_query(self, transfer_ids, timeout=None):
         """
 
         jobs = None
+        responses = {}
+        fts_session = requests.Session()
 
         if not isinstance(transfer_ids, list):
             transfer_ids = [transfer_ids]
 
-        responses = {}
-        fts_session = requests.Session()
-        xfer_ids = ','.join(transfer_ids)
-        jobs = fts_session.get('%s/jobs/%s?files=file_state,dest_surl,finish_time,start_time,staging_start,staging_finished,reason,source_surl,file_metadata' % (self.external_host, xfer_ids),
-                               verify=self.verify,
-                               cert=self.cert,
-                               headers=self.headers,
-                               timeout=timeout)
-
-        if jobs is None:
-            record_counter('transfertool.fts3.%s.bulk_query.failure' % self.__extract_host(self.external_host))
-            for transfer_id in transfer_ids:
-                responses[transfer_id] = Exception('Transfer information returns None: %s' % jobs)
-        elif jobs.status_code == 200 or jobs.status_code == 207:
-            try:
-                record_counter('transfertool.fts3.%s.bulk_query.success' % self.__extract_host(self.external_host))
-                labels = {'state': 'success', 'host': self.__extract_host(self.external_host)}
-                BULK_QUERY_COUNTER.labels(**labels).inc()
-                jobs_response = jobs.json()
-                responses = self.__bulk_query_responses(jobs_response)
-            except ReadTimeout as error:
-                raise TransferToolTimeout(error)
-            except JSONDecodeError as error:
-                raise TransferToolWrongAnswer(error)
-            except Exception as error:
-                raise Exception("Failed to parse the job response: %s, error: %s" % (str(jobs), str(error)))
+        # multi-VO mode
+        if self.multi_vo:
```

Comment:
same comment as previous: don't copy paste code move the common code to a function and call it where needed

Reply:
same as above

Reply:
method abstracted